octoprint_mrbeam/printing/profile.py

```diff
@@ -65,394 +73,252 @@ class InvalidProfileError(Exception):
         except InvalidProfileError:
             return None
 
     def remove(self, identifier):
-        if identifier == "_default":
+        # Overloaded from OP because of printerProfiles path (default_id)
+        if self._current is not None and self._current["id"] == identifier:
+            return False
+        elif settings().get(self.SETTINGS_PATH_PROFILE_DEFAULT_ID) == identifier:
             return False
         return self._remove_from_path(self._get_profile_path(identifier))
 
     def save(self, profile, allow_overwrite=False, make_default=False):
         """
         Saves given profile to file.
+        /!\ make_default uses the octoprint printerProfile path.
         :param profile:
         :param allow_overwrite:
         :param make_default:
         :return:
         """
-        if "id" in profile and profile["id"] != "":
-            identifier = profile["id"]
-        elif "name" in profile:
-            identifier = profile["name"]
-        else:
-            raise InvalidProfileError("profile must contain either id or name")
-
-        identifier = self._sanitize(identifier)
-        profile["id"] = identifier
-        profile = dict_clean(profile, self.__class__.default)
-
-        if identifier == "_default":
-            default_profile = dict_merge(self._load_default(), profile)
-            if not self._ensure_valid_profile(default_profile):
-                raise InvalidProfileError()
-
-            self.settings.set(
-                self.SETTINGS_PATH_PROFILE_DEFAULT_PROFILE,
-                default_profile,
-                defaults=dict(
-                    lasercutterprofiles=dict(defaultProfile=self.__class__.default)
-                ),
-            )
-            self.settings.save()
-        else:
-            self._save_to_path(
-                self._get_profile_path(identifier),
-                profile,
-                allow_overwrite=allow_overwrite,
-            )
-
-            if make_default:
-                self.set_default(identifier)
+        profile = PrinterProfileManager.save(
+            self, profile, allow_overwrite, make_default=False
+        )
+        # TODO : I don't really understand what this is for - investigate
+        # if identifier == "_default":
+        #     default_profile = dict_merge(self._load_default(), profile)
+        #     if not self._ensure_valid_profile(default_profile):
+        #         raise InvalidProfileError()
+
+        #     settings().set(
+        #         self.SETTINGS_PATH_PROFILE_DEFAULT_PROFILE,
+        #         default_profile,
+        #         defaults=dict(
+        #             lasercutterprofiles=dict(defaultProfile=LASER_PROFILE_DEFAULT)
+        #         ),
+        #     )
+        #     settings().save()
+        # else:
+        #     self._save_to_path(
+        #         self._get_profile_path(identifier),
+        #         profile,
+        #         allow_overwrite=allow_overwrite,
+        #     )
```

Comment:
@Gallore I believe this is part of the deprecated feature where you could modify the profile from the front end?